From 5633ce55ffd8e3df4a2c8c29c597b48aa7676435 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 16 Jan 2020 17:22:10 +0000
Subject: [PATCH] GCC: use scoped_refptr to return abstract VideoTrackInterface

Fix compilation issue in GCC with CrossThreadCopierPassThrough and VideoTrackInterface.

Bug: 819294
Change-Id: Id453c3a7bc63fc75a0f882af450730ada5b717de
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2002512
Reviewed-by: Chris Mumford <cmumford@google.com>
Reviewed-by: Guido Urdaneta <guidou@chromium.org>
Commit-Queue: Guido Urdaneta <guidou@chromium.org>
Cr-Commit-Position: refs/heads/master@{#732437}
---
 .../peerconnection/adapters/web_rtc_cross_thread_copier.h    | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/third_party/blink/renderer/modules/peerconnection/adapters/web_rtc_cross_thread_copier.h b/third_party/blink/renderer/modules/peerconnection/adapters/web_rtc_cross_thread_copier.h
index 54cb7d13bff3..0d6c40f0b428 100644
--- a/third_party/blink/renderer/modules/peerconnection/adapters/web_rtc_cross_thread_copier.h
+++ b/third_party/blink/renderer/modules/peerconnection/adapters/web_rtc_cross_thread_copier.h
@@ -165,8 +165,9 @@ struct CrossThreadCopier<blink::MediaStreamVideoTrack>
 };
 
 template <>
-struct CrossThreadCopier<webrtc::VideoTrackInterface>
-    : public CrossThreadCopierPassThrough<webrtc::VideoTrackInterface> {
+struct CrossThreadCopier<rtc::scoped_refptr<webrtc::VideoTrackInterface>>
+    : public CrossThreadCopierPassThrough<
+          rtc::scoped_refptr<webrtc::VideoTrackInterface>> {
   STATIC_ONLY(CrossThreadCopier);
 };
 
-- 
2.24.1

